src/www/components/popup/PopupComp.jsx

```diff
@@ -0,0 +1,183 @@
+import "./PopupComp.scss";
+import React, { Component } from "react";
+
+import Input from "../../components/input/Input";
+import axios from "axios";
+import Popup from "reactjs-popup";
+import ImageUploader from "react-images-upload";
+import ScrollArea from "react-scrollbar";
+import { Redirect } from "react-router-dom";
+import { withTranslation } from "react-i18next";
+
+class PopupComp extends Component {
+  state = {
+    teamName: "",
+    description: "",
+    teamCategories: [],
+    teamPicture: "",
+    serverResponse: "",
+    redirect: false,
+  };
+
+  createTeam = (name, description, categories, photo) => {
+    axios
+      .post("/api/teams/create", {
+        name,
+        description,
+        categories,
+        photo,
+      })
+      .then((res) => this.handleResponse(res.data))
+      .catch((err) => this.handleResponse(err.response.data));
+  };
+
+  handleResponse = (data) => {
+    const { t } = this.props;
+    if (data.error) {
+      if (data.error.includes("Team name")) {
+        this.setState({
+          serverResponse: t("createteam.nameerror"),
+        });
+      } else if (data.error.includes("Authentication")) {
+        this.setState({
+          serverResponse: t("createteam.authentication"),
+        });
+      }
+    } else {
+      this.setState({
+        redirect: true,
```

Comment:
Instead of setting `redirect` to true, you can set a string field `teamId` that's originally empty to the id that the server returns. This way, you can check if `this.state.teamId !== ""` in render() and if not, redirect to the page with that id.